pennylane/transforms/hamiltonian_expand.py

```diff
@@ -197,3 +201,202 @@ def processing_fn(res):
+
+    # Create the tapes, group observables if group==True
+    if group:
+        m_groups = _group_measurements(measurements)
+        # Update ``idxs_coeffs`` list such that it tracks the new ``m_groups`` list of lists
+        tmp_idxs = []
+        for m_group in m_groups:
+            if len(m_group) == 1:
+                tmp_idxs.append(idxs_coeffs[measurements.index(m_group[0])])
+            else:
+                tmp_idxs.append([idxs_coeffs[measurements.index(m)] for m in m_group])
+        idxs_coeffs = tmp_idxs
+        tapes = [
+            QuantumScript(ops=tape._ops, measurements=m_group, prep=tape._prep)
+            for m_group in m_groups
+        ]
+    else:
+        tapes = [
+            QuantumScript(ops=tape._ops, measurements=[m], prep=tape._prep) for m in measurements
+        ]
+
+    def processing_fn(expanded_results):
+        results = []  # [(m_idx, result)]
+        for tape_res, tape_idxs in zip(expanded_results, idxs_coeffs):
+            if isinstance(tape_idxs[0], tuple):  # tape_res contains only one result
+                if not qml.active_return():  # old return types
+                    tape_res = tape_res[0]
+                for idx, coeff in tape_idxs:
+                    results.append((idx, tape_res if coeff is None else coeff * tape_res))
+                continue
+            # tape_res contains multiple results
+            for res, idxs in zip(tape_res, tape_idxs):
+                for idx, coeff in idxs:
+                    results.append((idx, res if coeff is None else coeff * res))
+
+        # sum results by idx
+        res_dict = {}
+        for idx, res in results:
+            if idx in res_dict:
+                res_dict[idx] += res
+            else:
+                res_dict[idx] = res
+
+        # sort results by idx
+        results = [res_dict[key] for key in sorted(res_dict)]
+
+        return results[0] if len(results) == 1 else results
+
+    return tapes, processing_fn
+
+
+def _group_measurements(measurements: List[MeasurementProcess]) -> List[List[MeasurementProcess]]:
+    """Group observables of measurements list into qubit-wise commuting groups.
+
+    Args:
+        measurements (List[MeasurementProcess]): list of measurement processes
+
+    Returns:
+        List[List[MeasurementProcess]]: list of qubit-wise commuting measurement groups
+    """
+    qwc_groups = []
+    for m in measurements:
+        if len(m.wires) == 0:  # measurement acts on all wires: e.g. qml.counts()
+            qwc_groups.append((m.wires, [m]))
+            continue
+
+        op_added = False
+        for idx, (wires, group) in enumerate(qwc_groups):
+            if len(wires) > 0 and all(wire not in m.wires for wire in wires):
+                qwc_groups[idx] = (wires + m.wires, group + [m])
+                op_added = True
+                break
+
+        if not op_added:
+            qwc_groups.append((m.wires, [m]))
+
+    return [group[1] for group in qwc_groups]
```

Comment:
Maybe I am misunderstanding something, but I think this function is not doing what you want it to do:
When you run the following example:
```python
measurements = [qml.expval(qml.PauliX(0) @ qml.PauliX(1)), qml.expval(qml.PauliY(0) @ qml.PauliY(1)), qml.expval(qml.PauliX(0)), qml.expval(qml.PauliY(0))]
>>> _group_measurements(measurements)
[[expval(PauliX(wires=[0]) @ PauliX(wires=[1]))],
 [expval(PauliY(wires=[0]) @ PauliY(wires=[1]))],
 [expval(PauliX(wires=[0]))],
 [expval(PauliY(wires=[0]))]]
```
It yields 4 groups. However, it should be only two groups. You can just use the already implemented `qml.grouping.group_observables` functionality:

```python
>>> qml.grouping.group_observables([_.obs for _ in measurements])
[[PauliX(wires=[0]) @ PauliX(wires=[1]), PauliX(wires=[0])],
 [PauliY(wires=[0]) @ PauliY(wires=[1]), PauliY(wires=[0])]]
```

Reply:
@Qottmann, This I think was meant to do very trivial grouping as the grouping function is being updated separately. Once that is fully updated, we can swap this grouping function out for that one 👍🏼

Reply:
Ah okay so this is known?

If we are going to advertise `sum_expand` for the new release I would either use the old grouping function and mark it with a TODO for later once the Pauli update has been merged, or leave the grouping part all together. At the moment it is advertising something that it is not doing so I think it would just create confusion.

Reply:
This is a private function, thus it should NOT be advertised.

Reply:
I meant the overall `group=True` pipeline

Reply:
But you are right, the docstring here is wrong. This method only groups observables that act on non-overlapping wires.

Changed the docstring to make it more clear.